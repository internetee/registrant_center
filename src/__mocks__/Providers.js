import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import translations from '../translations';
import state from './mockReduxState';

const lang = 'et';

export default function Providers({ children }) {
    return (
        <BrowserRouter>
            <Provider store={state}>
                <CookiesProvider>
                    <IntlProvider
                        key={lang}
                        defaultLocale="et"
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}
