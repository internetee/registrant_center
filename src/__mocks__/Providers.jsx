import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router';
import translations from '../translations';
import PropTypes from 'prop-types';

const lang = 'et';

export default function Providers({ children, store }) {
    return (
        <BrowserRouter>
            <Provider store={store}>
                <CookiesProvider>
                    <IntlProvider
                        defaultLocale="et"
                        key={lang}
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}

Providers.propTypes = {
    children: PropTypes.node.isRequired,
    store: PropTypes.object.isRequired,
};
