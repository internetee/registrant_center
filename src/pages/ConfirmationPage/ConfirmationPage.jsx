import { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';
import { Button, Container, Table } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { Loading, MainLayout, PageMessage, MessageModule } from '../../components';
import {
    fetchVerification as fetchVerificationAction,
    respondToVerification as respondToVerificationAction,
} from '../../redux/reducers/verification';
import { useParams } from 'react-router-dom';
import { should } from 'vitest';

const ConfirmationPage = ({
    ui,
    message,
    fetchVerification,
    respondToVerification,
    verification,
}) => {
    const { name, type, token } = useParams();
    const [isLoading, setIsLoading] = useState(true);
    const { uiElemSize } = ui;

    if (!name || !type) {
        return (
            <MainLayout titleKey="errorpage.title">
                <div className="page">
                    <PageMessage
                        headerContent={<FormattedMessage id="errorpage.none.message.title" />}
                        icon="times circle outline"
                    />
                </div>
            </MainLayout>
        );
    }

    const headerAltText = () => {
        let str = '';
        if (type === 'change') {
            if (verification.status === 'confirmed') {
                str = 'confirmation.change_confirmed_alt';
            } else {
                str = 'confirmation.change_rejected_alt';
            }
        } else if (type === 'delete') {
            if (verification.status === 'confirmed') {
                str = 'confirmation.delete_confirmed_alt';
            } else {
                str = 'confirmation.delete_confirmed_alt';
            }
        }

        return str;
    };

    const handleVerification = async (shouldConfirm) => {
        setIsLoading(true);
        await respondToVerification(
            name,
            token,
            shouldConfirm,
            type
        );
        setIsLoading(false);
    };

    useEffect(() => {
        (async () => {
            if (!token) {
                // If no token, just show the form without fetching
                setIsLoading(false);
                return;
            }

            setIsLoading(true);
            await fetchVerification({
                domain: name,
                token: token,
                type: type,
            });
            setIsLoading(false);
        })();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [fetchVerification]);

    if (isLoading) return <Loading />;

    return (
        <MainLayout
            hasBackButton
            titleKey={
                type === 'change'
                    ? 'confirmation.change_title'
                    : 'confirmation.delete_title'
            }
        >
            {!isLoading && message && <MessageModule message={message} />}
            <div className="page page--whois">
                {verification.domainName != null ? (
                    <>
                        {verification.status == null ? (
                            <div className="page--whosis">
                                <div className="page--header">
                                    <Container textAlign="center">
                                        <div>
                                            <h2>{verification.domainName}</h2>
                                            <p>
                                                <FormattedMessage
                                                    id={
                                                        type === 'change'
                                                            ? 'confirmation.change_default_alt'
                                                            : 'confirmation.delete_default_alt'
                                                    }
                                                />
                                            </p>
                                        </div>
                                    </Container>
                                </div>
                                {type === 'change' ? (
                                    <div className="page--block">
                                        {' '}
                                        <Container text>
                                            <header className="page--block--header">
                                                <h2>
                                                    <FormattedMessage id="confirmation.new_registrant" />
                                                </h2>
                                            </header>
                                            <Table basic="very">
                                                <Table.Body>
                                                    <Table.Row>
                                                        <Table.Cell width="4">
                                                            <FormattedMessage
                                                                id="domain.registrar.name"
                                                                tagName="strong"
                                                            />
                                                        </Table.Cell>
                                                        <Table.Cell>
                                                            {verification.newRegistrant.name}
                                                        </Table.Cell>
                                                    </Table.Row>
                                                    <Table.Row>
                                                        <Table.Cell width="4">
                                                            <FormattedMessage
                                                                id="domain.registrant.ident"
                                                                tagName="strong"
                                                            />
                                                        </Table.Cell>
                                                        <Table.Cell>
                                                            {verification.newRegistrant.ident}
                                                        </Table.Cell>
                                                    </Table.Row>
                                                    <Table.Row>
                                                        <Table.Cell width="4">
                                                            <FormattedMessage
                                                                id="domain.contact.country"
                                                                tagName="strong"
                                                            />
                                                        </Table.Cell>
                                                        <Table.Cell>
                                                            {verification.newRegistrant.country}
                                                        </Table.Cell>
                                                    </Table.Row>
                                                </Table.Body>
                                            </Table>
                                        </Container>
                                    </div>
                                ) : null}
                                <div className="page--block">
                                    <Container text>
                                        <header className="page--block--header">
                                            <h2>
                                                <FormattedMessage id="confirmation.current_registrant" />
                                            </h2>
                                        </header>
                                        <Table basic="very">
                                            <Table.Body>
                                                <Table.Row>
                                                    <Table.Cell width="4">
                                                        <FormattedMessage
                                                            id="domain.registrar.name"
                                                            tagName="strong"
                                                        />
                                                    </Table.Cell>
                                                    <Table.Cell>
                                                        {verification.currentRegistrant.name}
                                                    </Table.Cell>
                                                </Table.Row>
                                                <Table.Row>
                                                    <Table.Cell width="4">
                                                        <FormattedMessage
                                                            id="domain.registrant.ident"
                                                            tagName="strong"
                                                        />
                                                    </Table.Cell>
                                                    <Table.Cell>
                                                        {verification.currentRegistrant.ident}
                                                    </Table.Cell>
                                                </Table.Row>
                                                <Table.Row>
                                                    <Table.Cell width="4">
                                                        <FormattedMessage
                                                            id="domain.contact.country"
                                                            tagName="strong"
                                                        />
                                                    </Table.Cell>
                                                    <Table.Cell>
                                                        {verification.currentRegistrant.country}
                                                    </Table.Cell>
                                                </Table.Row>
                                            </Table.Body>
                                        </Table>
                                    </Container>
                                </div>
                                <div className="page--block">
                                    <Container textAlign="center">
                                        <div className="page--header--actions">
                                            <Button
                                                data-test="reject"
                                                disabled={isLoading}
                                                loading={isLoading}
                                                onClick={() => handleVerification('rejected')}
                                                secondary
                                                size={uiElemSize}
                                            >
                                                <FormattedMessage id="confirmation.reject" />
                                            </Button>
                                            <Button
                                                data-test="confirm"
                                                disabled={isLoading}
                                                loading={isLoading}
                                                onClick={() => handleVerification('confirmed')}
                                                primary
                                                size={uiElemSize}
                                            >
                                                <FormattedMessage id="confirmation.confirm" />
                                            </Button>
                                        </div>
                                    </Container>
                                </div>
                            </div>
                        ) : (
                            <div className="page--whosis">
                                <div className="page--header">
                                    <Container textAlign="center">
                                        <div>
                                            <h2>{verification.domainName}</h2>
                                            <p>
                                                <FormattedMessage id={headerAltText()} />
                                            </p>
                                        </div>
                                    </Container>
                                </div>
                                <div className="page--block">
                                    <Container text>
                                        <header className="page--block--header">
                                            <h2>
                                                <FormattedMessage id="confirmation.valid_registrant" />
                                            </h2>
                                        </header>
                                        <Table basic="very">
                                            <Table.Body>
                                                <Table.Row>
                                                    <Table.Cell width="4">
                                                        <FormattedMessage
                                                            id="domain.registrar.name"
                                                            tagName="strong"
                                                        />
                                                    </Table.Cell>
                                                    <Table.Cell>
                                                        {verification.currentRegistrant.name}
                                                    </Table.Cell>
                                                </Table.Row>
                                                <Table.Row>
                                                    <Table.Cell width="4">
                                                        <FormattedMessage
                                                            id="domain.registrant.ident"
                                                            tagName="strong"
                                                        />
                                                    </Table.Cell>
                                                    <Table.Cell>
                                                        {verification.currentRegistrant.ident}
                                                    </Table.Cell>
                                                </Table.Row>
                                                <Table.Row>
                                                    <Table.Cell width="4">
                                                        <FormattedMessage
                                                            id="domain.contact.country"
                                                            tagName="strong"
                                                        />
                                                    </Table.Cell>
                                                    <Table.Cell>
                                                        {verification.currentRegistrant.country}
                                                    </Table.Cell>
                                                </Table.Row>
                                            </Table.Body>
                                        </Table>
                                    </Container>
                                </div>
                            </div>
                        )}
                    </>
                ) : (
                    <PageMessage
                        headerContent={
                            <FormattedMessage
                                id={
                                    type === 'change'
                                        ? 'confirmation.change_not_available'
                                        : 'confirmation.delete_not_available'
                                }
                            />
                        }
                        icon="frown outline"
                    />
                )}
            </div>
        </MainLayout>
    );
};

export default connect(
    (state) => ({
        ui: state.ui,
        message: state.message,
        verification: state.verification,
    }),
    (dispatch) =>
        bindActionCreators(
            {
                fetchVerification: fetchVerificationAction,
                respondToVerification: respondToVerificationAction,
            },
            dispatch
        )
)(ConfirmationPage);

ConfirmationPage.propTypes = {
    ui: PropTypes.object.isRequired,
    message: PropTypes.string,
    fetchVerification: PropTypes.func.isRequired,
    verification: PropTypes.object.isRequired,
};

ConfirmationPage.defaultProps = {
    message: '',
};
