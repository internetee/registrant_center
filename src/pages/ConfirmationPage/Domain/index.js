import React, { useState } from 'react';
import { Icon, Table } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import classNames from 'classnames';
import { WhoIsEdit } from '../../../components';

export default function Domain({ id, name, contacts, onChange }) {
    const [isOpen, setIsOpen] = useState(false);
    return (
        <Table.Row verticalAlign="top">
            <Table.Cell width={6}>
                <Link to={`/domain/${id}`}>{name}</Link>
            </Table.Cell>
            <Table.Cell width={8}>
                <div className="ui form">
                    <WhoIsEdit checkAll contacts={contacts} isOpen={isOpen} onChange={onChange} />
                </div>
            </Table.Cell>
            <Table.Cell singleLine textAlign="right" width={2}>
                <button onClick={() => setIsOpen(!isOpen)} type="button">
                    <FormattedMessage id="whois.domain.detailInfo" />
                    <Icon
                        className={classNames({ 'vertically flipped': isOpen })}
                        name="angle down"
                    />
                </button>
            </Table.Cell>
        </Table.Row>
    );
}
