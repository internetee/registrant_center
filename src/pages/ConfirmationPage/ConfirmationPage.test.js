import 'core-js';
import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationPage from './ConfirmationPage';
import Providers from '../../__mocks__/Providers';

describe('pages/Home', () => {
	it('should render content', () => {
		const component = shallow(
			<Providers>
				<ConfirmationPage />
			</Providers>
		);
		expect(component).toMatchSnapshot();
	});
});
