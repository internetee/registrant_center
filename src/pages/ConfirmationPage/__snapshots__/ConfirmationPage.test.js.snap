// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home should render content 1`] = `ShallowWrapper {}`;
